backend/src/main/java/gov/cdc/usds/simplereport/config/scalars/datetime/DateTimeScalarCoercion.java

```diff
@@ -65,10 +70,12 @@ private LocalDateTime getLocalDateTime(Date input) {
   }
 
   @Override
-  public Object serialize(Object dataFetcherResult) throws CoercingSerializeException {
-    if (dataFetcherResult == null) {
-      throw new CoercingSerializeException("Unable to serialize null value");
-    } else if (dataFetcherResult instanceof Date) {
+  public Object serialize(
+      @NotNull Object dataFetcherResult,
+      @NotNull GraphQLContext graphQLContext,
+      @NotNull Locale locale)
+      throws CoercingSerializeException {
```

Comment:
Previous overridden method signature was deprecated in favor of this.

Reply:
Question for my own understanding... (I have never upgraded Spring before 😅) How did you know this method should now look like this? Was it trial and error or is there some resource you are taking this from? 🤔

Reply:
IIRC, I was mainly alerted through Intellij after updating the dependency related to the change. e.g. after updating the dependency but before updating the code, it looked like this: 
<img width="884" alt="Screenshot 2024-12-04 at 1 46 24 PM" src="https://github.com/user-attachments/assets/f39dbe5a-bbde-4a2d-ad43-68d650896e57">
So then after seeing that, I'd look into the Coercing interface for what should replace it. Luckily there are usually replacements for deprecated methods and they're not just wiped out entirely.

Another Intellij trick is to right click on the root folder, then `Analyze -> Run Inspection by Name` and type in "deprecated" and then click on the one for Java:

<img width="670" alt="Screenshot 2024-12-04 at 2 00 39 PM" src="https://github.com/user-attachments/assets/859214c8-8336-4fa9-8b03-f7a94501397a">

Then it'll show you anything that's deprecated or that altogether doesn't work anymore 🧐

Reply:
Thank you for the detailed explanation and going over this in eng sync as well 🙌